pkg/ck8s/workload_cluster.go

```diff
@@ -226,17 +240,91 @@ func (w *Workload) RefreshCertificates(ctx context.Context, machine *clusterv1.M
 		return 0, fmt.Errorf("failed to refresh certificates: %w", err)
 	}
 
+	return planResponse.Seed, nil
+}
+
+func (w *Workload) refreshCertificatesRun(ctx context.Context, machine *clusterv1.Machine, nodeToken string, request *apiv1.ClusterAPICertificatesRunRequest) (int, error) {
+	runResponse := &apiv1.ClusterAPICertificatesRunResponse{}
+	header := w.newHeaderWithNodeToken(nodeToken)
+
+	k8sdProxy, err := w.GetK8sdProxyForMachine(ctx, machine)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create k8sd proxy: %w", err)
+	}
+
+	if err := w.doK8sdRequest(ctx, k8sdProxy, http.MethodPost, "1.0/x/capi/refresh-certs/run", header, request, runResponse); err != nil {
```

Comment:
Maybe we can also use `k8s-snap-api` here as well?

Reply:
https://github.com/canonical/cluster-api-k8s/pull/65#discussion_r1805602467